tests/testthat/test_schematic_rest_api.R

```diff
@@ -73,74 +160,304 @@ test_that("model_submit successfully uploads to synapse", {
+    )
+  
+  # FIXME
+  # if source component is errant, function seems to run endlessly?
+  
+  # reqs <- model_component_requirements(
+  #   url = file.path(schematic_url, "v1/model/component-requirements"),
+  #   schema_url = "https://raw.githubusercontent.com/ncihtan/data-models/main/HTAN.model.jsonld",
+  #   source_component = "xxx",
+  #   as_graph = FALSE)
+})
 
-test_that("get_asset_view_table returns asset view table", {
+test_that("manifest_download returns a manifest as a dataframe", {
   skip_it()
-  av <- get_asset_view_table(url=file.path(schematic_url, "v1/storage/assets/tables"),
-                             access_token = Sys.getenv("SYNAPSE_PAT"),
-                       asset_view="syn23643253")
-  storage_tbl <- subset(av, av$name == "synapse_storage_manifest.csv")
-  expect_true(inherits(av, "data.frame"), "name" %in% names(av))
+  
+ manifest <- manifest_download(
+   url = file.path(schematic_url, "v1/manifest/download"),
+   manifest_id = "syn51078535",
+   access_token = Sys.getenv("SYNAPSE_PAT")
+   )
+ 
+  exp <- setNames(c("BulkRNA-seqAssay", "CSV/TSV", "Sample_A", "GRCm38", NA, 2022L, "syn28278954"),
+    c("Component", "File Format", "Filename", "Genome Build", "Genome FASTA", "Sample ID", "entityId"))
+  
+  expect_equal(unlist(manifest), exp)
+  expect_true(is.data.frame(manifest))
+  expect_true(nrow(manifest) == 1)
 })
 
-test_that("asset_tables returns a data.frame", {
+test_that("manifest_download errors as expected", {
   skip_it()
-  tst <- get_asset_view_table(url=file.path(schematic_url, "v1/storage/assets/tables"),
-                       asset_view = "syn28559058",
-                       access_token = Sys.getenv("SYNAPSE_PAT"),
-                       return_type="json")
-  expect_identical(nrow(tst), 4L)
   
-  expect_error(get_asset_view_table(url=file.path(schematic_url, "v1/storage/assets/tables"),
-                                  asset_view = "syn28559058",
-                                  access_token = Sys.getenv("SYNAPSE_PAT"),
-                                  return_type = "csv")
-               )
+  # errant URL DOESNT ERROR
+  # outputs a list including status and detail message
+  error <- manifest_download(
+    url = file.path(schematic_url, "v1/manifest/download/TYPO"),
+    manifest_id = "syn51078535",
+    access_token = Sys.getenv("SYNAPSE_PAT")
+  )
+  
+  expect_true(is.list(error))
+  expect_true(error$status == 404)
+  
+  # bad manifest id
+  expect_error(
+    manifest_download(
+      url = file.path(schematic_url, "v1/manifest/download"),
+      manifest_id = "syn51078535xx",
+      access_token = Sys.getenv("SYNAPSE_PAT")
+    )
+  )
+  
+  # bad PAT
+  expect_error(
+    manifest_download(
+      url = file.path(schematic_url, "v1/manifest/download"),
+      manifest_id = "syn51078535",
+      access_token = "bad PAT"
+    )
+  )
+})
+
+test_that("get_asset_view_table returns output as expected", {
+  skip_it()
+  
+  # returns manifest from csv
```

Comment:
@afwillia another behavior that made me 🤔 . We could just force the function to use `return_type = json` if that's all thats required rn.